doc/source/cluster/slurm.rst

```diff
@@ -21,51 +21,40 @@ Here are some community-contributed templates for using SLURM with Ray:
 
 .. _`Template script`: https://gist.github.com/pengzhenghao/b348db1075101a9b986c4cdfea13dcd6
 
+
 Starter SLURM script
 --------------------
 
 .. code-block:: bash
 
   #!/bin/bash
-
   #SBATCH --job-name=test
   #SBATCH --cpus-per-task=5
   #SBATCH --mem-per-cpu=1GB
-  #SBATCH --nodes=3
-  #SBATCH --tasks-per-node 1
-
-  worker_num=2 # Must be one less that the total number of nodes
+  #SBATCH --nodes=4
+  #SBATCH --tasks-per-node=1
+  #SBATCH --time=00:30:00
+  #SBATCH --reservation=test
 
-  # module load Langs/Python/3.6.4 # This will vary depending on your environment
-  # source venv/bin/activate
+  let "worker_num=(${SLURM_NTASKS} - 1)"
 
-  nodes=$(scontrol show hostnames $SLURM_JOB_NODELIST) # Getting the node names
-  nodes_array=( $nodes )
-
-  node1=${nodes_array[0]}
-
-  ip_prefix=$(srun --nodes=1 --ntasks=1 -w $node1 hostname --ip-address) # Making address
-  suffix=':6379'
-  ip_head=$ip_prefix$suffix
-  redis_password=$(uuidgen)
+  # Define the total number of CPU cores available to ray
+  let "total_cores=${worker_num} * ${SLURM_CPUS_PER_TASK}"
 
+  suffix='6379'
+  ip_head=`hostname`:$suffix
   export ip_head # Exporting for latter access by trainer.py
 
-  srun --nodes=1 --ntasks=1 -w $node1 ray start --block --head --redis-port=6379 --redis-password=$redis_password & # Starting the head
+  srun --nodes=1 --ntasks=1 --cpus-per-task=${SLURM_CPUS_PER_TASK} --nodelist=`hostname` ray start --head --block --dashboard-host 0.0.0.0 --port=6379 &
```

Comment:
could you add a quick comment about what this does? this would be super helpful to other readers